lib/utils/ember.js

```diff
@@ -527,3 +529,13 @@ function isEmberObjectImplementingUnknownProperty(node) {
   }
   return false;
 }
+
+function isObserverDecorator(node) {
+  assert(types.isDecorator(node), 'Should only call this function on a Decorator');
+  return (
+    types.isDecorator(node) &&
+    types.isCallExpression(node.expression) &&
+    types.isIdentifier(node.expression.callee) &&
+    node.expression.callee.name === 'observes'
```

Comment:
I know this isn't a new issue, but we really should tie these callee name comparisons to specific imports.

For example, add the following as a "bad" test case (should fail, but I suspect it will pass at the moment):

```js
import EmberObject from '@ember/object';
import { observes as o } from '@ember-decorators/object';

class Foo extends EmberObject {
  @o('lol')
  doSomething() {}
}
```

---

Since this isn't a new issue, it may be useful to land this change as is and make a meta issue about fixing the various utils we have to be more intelligent about scoping.

Reply:
Totally agreed! We've been getting better about import validation recently thanks to one of our new utils `getSourceModuleNameForIdentifier` but that util isn't sufficient in this case. I opened #590 to track improving our import validation.